org-code-javabuilder/playground/src/main/java/org/code/playground/Board.java

```diff
@@ -60,28 +67,66 @@ public void setBackgroundImage(String filename) throws FileNotFoundException {}
    *
    * @param image the image to add. If the image is already on the board, this method does nothing.
    */
-  public void addClickableImage(ClickableImage image) {}
+  public void addClickableImage(ClickableImage image) {
+    if (this.clickableImages.containsKey(image.getId())) {
+      return;
+    }
+    HashMap<String, String> details = image.getDetails();
+    this.addIndexToDetails(details);
+    this.clickableImages.put(image.getId(), image);
+    this.playgroundMessageHandler.sendMessage(
+        new PlaygroundMessage(PlaygroundSignalKey.ADD_CLICKABLE_ITEM, details));
```

Comment:
Could we reuse `addItemHelper` here too if we passed it the list to add to or an `isClickable` boolean maybe?

Reply:
Since the `clickableImages` hash needs to have `<String, ClickableImage>` as the type instead of `<String, Item>`, it would end up requiring doing some if/else that doesn't seem super clean to me. I think I will leave it as is since it only ends up being 3 repeat lines of code.

Reply:
Ah good point, sounds good